File: examples/qr-login.ts
import { Wazeko } from "../packages/wazeko/src/index.js";
import { QrCodeEvent, ConnectionState, Jid } from "../packages/wazeko-types/src/index.js";

async function main() {
  const client = Wazeko.builder()
    .authStore("./auth_qr")
    .authMethod("qr")
    .printQr(true)
    .build();

  client.on("qr", (qr: QrCodeEvent) => {
    console.log(`New QR received, attempt #${qr.attempts}`);
  });

  client.on("connection.update", (state: ConnectionState) => {
    console.log(`Connection state: ${state}`);
  });

  client.on("authenticated", ({ userJid }: { userJid: Jid }) => {
    console.log("Successfully logged in as:", userJid);
  });

  await client.connect();
}

main().catch(console.error);

File: crates/wazeko/examples/qr_login.rs
use wazeko::{AuthMethod, Event, Wazeko};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    tracing_subscriber::fmt::init();

    let client = Wazeko::builder()
        .auth_store("./auth_qr_demo")
        .auth_method(AuthMethod::QrCode)
        .print_qr(true)
        .build();

    client.connect().await?;

    let mut events = client.events();
    while let Some(event) = events.recv().await {
        match event {
            Event::Qr(qr_event) => {
                println!("New QR challenge received! Attempt #{}", qr_event.attempts);
            }
            Event::ConnectionUpdate(state) => {
                println!("Connection state changed: {state:?}");
            }
            Event::Authenticated { user_jid } => {
                println!("Successfully authenticated as: {user_jid}");
                break;
            }
            _ => {}
        }
    }

    Ok(())
}
